### ISSUE TITLE
0.92 and problems 

### ISSUE THREAD

**Issue by sagitt** [hunterjm/hassio-addons#43]
After update i see this!

Integration xboxone not found when trying to verify its media_player platform.

**Comment by XelandOne** [comment]
I ran into the same problem with HA 0.92 installed. It probably has something to do with the new way of how Integrations are structured (https://developers.home-assistant.io/blog/2019/04/12/new-integration-structure.html).
Should the path be /config/custom_components/xboxone/media_player.py/
or
/config/custom_components/media_player/xboxone/
?

**Comment by sagitt** [comment]
Maybe “create an empty __init__.py file ”

**Comment by sagitt** [comment]
Yes creating the empty init file all works...... please update this plugin for this new standard.

**Comment by XelandOne** [comment]
Perfect, thank you for testing. Do you use
/config/custom_components/xboxone/media_player.py/
or
/config/custom_components/media_player/xboxone/
and how does your configuration.yaml entry looks like?

**Comment by mirkochip88** [comment]
I've made a pull request

**Comment by SeraphimSerapis** [comment]
The component should also define a `DOMAIN` as listed [here](https://developers.home-assistant.io/docs/en/creating_component_index.html).

Furthermore, state updates are handled differently [now](https://goo.gl/Nvioub):
```
Updating state for media_player.xbox_one (<class 'custom_components.xboxone.media_player.XboxOneDevice'>) took 0.699 seconds. Please report platform to the developers at https://goo.gl/Nvioub
```

**Comment by GsynthChris** [comment]
> Yes creating the empty init file all works...... please update this plugin for this new standard.

where do we drop the empty init.py file?

**Comment by sagitt** [comment]
> > Yes creating the empty init file all works...... please update this plugin for this new standard.
> 
> where do we drop the empty init.py file?

in custom_components/xboxone/

**Comment by GsynthChris** [comment]
hmm. thats what I thought. I tried that and I still get an error. I tried init.py and even __init__.py cause all the other ones look like that hahaha

"Integration xboxone not found when trying to verify its media_player platform."

**Comment by SeraphimSerapis** [comment]
It needs to be __init__.py

**Comment by GsynthChris** [comment]
I tried `init.py` and even `__init__.py` and neither resolved it.  I am still getting `ERROR (MainThread) [homeassistant.components.hassio] Integration xboxone not found when trying to verify its media_player platform.`  and the media player isn't showing up.

:-(

**Comment by SeraphimSerapis** [comment]
Check out my comment above. You may have to define a domain, too. Just use `xbox_one` as value.

**Comment by mirkochip88** [comment]
To me __init__.py works.

**Comment by GsynthChris** [comment]
I tried the domain thing and a blank `__init__.py`

I wish I know what the heck I was doing wrong. :-(

**Comment by GsynthChris** [comment]
I figured it out!
According to this post: https://github.com/home-assistant/home-assistant/issues/23418
you need to change any `SUPPORT_VOLUME_STEP` in your media_player.py to `SUPPORT_VOLUME_SET`

I was so skeptical but as soon as I did it I checked config and it passed. I restarted and everything started working. YAYAYAYAYAYAYAY!!!!!

**Comment by SeraphimSerapis** [comment]
Interesting -- I didn't make these changes and it runs flawlessly on 0.92.1 for me. Will see if I can change volume when I am back home.

**Comment by GsynthChris** [comment]
yeah idk why we had different experiences. very strange!

**Comment by mbuist** [comment]
This is what I did to get it to work with 0.92.2
Edit the file media_player.py and Change  any SUPPORT_VOLUME_STEP in 
Add an Empty__init__.py in the same directory
Add manifest.json with the following content to the same directory

{
  "domain": "xboxone",
  "name": "Xboxone",
  "documentation": "https://github.com/hunterjm/hassio-addons",
  "dependencies": [],
  "codeowners": [],
  "requirements": []
}

Restarted Home assistant after checking config validation. It all started to work again

**Comment by tierant5** [comment]
I can confirm adding the blank `__init__.py` file under `custom_components/xboxone` and putting `DOMAIN = 'xbox_one'` inside the `media_player.py` file gets the component working again in 0.92.2. Thank you @SeraphimSerapis !

**Comment by michaelblight** [comment]
I was confused about the `media_player.py` reference. For anyone else still struggling like me, you need to create 3 files:

- `custom_components/xboxone/__init__.py` with no content.
- `custom_components/xboxone/media_player.py` with the content copied from the add-on's `xboxone.py`. And then change the 3 occurrences of `SUPPORT_VOLUME_STEP` to `SUPPORT_VOLUME_SET` and add another line with `DOMAIN = 'xbox_one'` (I put it at line 40).
- `custom_components/xboxone/manifest.json` with content below
```
{
  "domain": "xboxone",
  "name": "Xboxone",
  "documentation": "https://github.com/hunterjm/hassio-addons",
  "dependencies": [],
  "codeowners": [],
  "requirements": []
}
```
You should then be able to "Check config" and have it pass.

**Comment by jackkitley** [comment]
Is this still valid with latest version? 0.98.5

**Comment by tibzz42** [comment]
Hi there. This fix doesnt seems to work anymore on 0.100.3... Does anyone find a way to solve it ?
I try to apply what @michaelblight said, but i still have the error message when check the configuration file :(

**Comment by ccoombes** [comment]
Just tried this on 0.103.5, will all the above fixes I'm still getting the "Platform error media_player.xboxone - Integration 'xboxone' not found." error. Does anyone have a fix for this?
